### PR TITLE
feat: add support for create and delete foreign key constraint from model

### PR DESCRIPTION
**Pull request by elv1s**
### Description
Support adding and removing foreign key constraints.

Support for adding and removing foreign key constraints. I marked this as WIP since there are a couple of TODO items.
- [x] support FK table update
- [x] fix camelCase issue for foreign key ids

 I wanted to get some feedback in the meantime.

#### Related issues

- connect to https://github.com/strongloop/loopback-connector-postgresql/issues/348

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] Documentation in README.md was updated

### CODE REVIEW

File: lib/migration.js [hunk 1]
```diff
@@ -99,15 +99,36 @@ function mixinMigration(PostgreSQL) {
...
-      function(cb) {
-        self.addIndexes(model, actualIndexes, cb);
-      },
-    ], function(err, result) {
-      cb(err, result[0]);
+    self.discoverForeignKeys(self.table(model), {}, function(err, actualFks) {
+      if (err) {
+        console.log('Failed to discover "' + self.table(model) +
+        '" foreign keys', err);
+      }
```
Comment (raymondfeng) [inline lib/migration.js]: Should we callback with `err` and return here?

File: lib/migration.js [hunk 2]
```diff
@@ -99,15 +99,36 @@ function mixinMigration(PostgreSQL) {
...
-      },
-      function(cb) {
-        self.addIndexes(model, actualIndexes, cb);
-      },
-    ], function(err, result) {
-      cb(err, result[0]);
+    self.discoverForeignKeys(self.table(model), {}, function(err, actualFks) {
+      if (err) {
+        console.log('Failed to discover "' + self.table(model) +
+        '" foreign keys', err);
```
Comment (raymondfeng) [inline lib/migration.js]: Use `debug` instead of `console.log`?

Comment (elv1s) [reply lib/migration.js]: done

File: lib/migration.js (hunk 1 above)
Comment (elv1s) [reply lib/migration.js]: yes, I thought it might be aggressive to fail for an error looking up FKs, but I will cb with the error

**Comment by elv1s** [comment]
I'm taking a look at the test failure caused by failure to delete table with FK constraint. This happens inconsistently on my local as well.

There is a test failure which looks like it is coming from datasource-juggler `/loopback-datasource-juggler/test/include.test.js:81:14`. Anyone know what might be causing this? This happens in `master`
```
1) include
       should report errors if the PK is excluded:
     Uncaught AssertionError: expected null to exist
```

**Comment by dhmlau** [comment]
Seems like something that @b-admike is trying to fix in PR #1682 and #1683.

**Comment by elv1s** [comment]
looks like the test is stuck...

All the tests passed locally after pulling in PR #1682 from @b-admike

**Comment by dhmlau** [comment]
@slnode test please

**Comment by dhmlau** [comment]
According to @rmg, there was some infrastructure issue regarding CLA.  Let me kick off the tests again. If it's still stuck, we'll need help from @rmg.

**Comment by elv1s** [comment]
Looks like the CLA check is still stuck

**Comment by elv1s** [comment]
All checks are passing now 👍

**Comment by raymondfeng** [comment]
@elv1s Please squash your commits into one to make history clean with `git rebase -i HEAD~6`.

**Comment by dhmlau** [comment]
@slnode test please

**Comment by dhmlau** [comment]
@slnode test please
(let's run it again)

**Comment by elv1s** [comment]
Connected issue #348 can be closed.

When will this feature be published?

**Comment by dhmlau** [comment]
Let me publish now.

**Comment by dhmlau** [comment]
Done. `loopback-connector-postgresql@3.5.1` is published.

**Pull request merged**